web/elm/src/SideBar/SideBar.elm

```diff
@@ -96,7 +96,16 @@ update message model =
         Click SideBarResizeHandle ->
             ( { model | draggingSideBar = True }, [] )
 
-        Click (SideBarStarIcon pipelineID) ->
+        Click (SideBarFavoritedIcon pipelineID) ->
+            let
+                favoritedPipelines =
+                    toggle pipelineID model.favoritedPipelines
```

Comment:
i never looked at the toggle function before, i like it.

Reply:
And it likes you!